Library/Homebrew/archive.rb

```diff
@@ -0,0 +1,185 @@
+# typed: false
+# frozen_string_literal: true
+
+require "digest/md5"
+require "utils/curl"
+
+# Archive API client.
+#
+# @api private
+class Archive
+  extend T::Sig
+
+  include Context
+  include Utils::Curl
+
+  class Error < RuntimeError
+  end
+
+  sig { returns(String) }
+  def inspect
+    "#<Archive: item=#{@archive_item}>"
+  end
+
+  sig { params(item: T.nilable(String)).void }
+  def initialize(item: "homebrew")
+    @archive_item = item
+
+    raise UsageError, "Must set the Archive item!" unless @archive_item
+
+    ENV["HOMEBREW_FORCE_HOMEBREW_ON_LINUX"] = "1" if @archive_item == "homebrew" && !OS.mac?
```

Comment:
Can you add a comment explaining this?

Reply:
This line is here on the foundational software engineering principles of cargo cult engineering and [Chesterton's Fence](https://en.wikipedia.org/wiki/Wikipedia:Chesterton's_fence), cribbed from https://github.com/Homebrew/brew/blob/3efab3efa7114afe713b2485032b740e6b55d6e4/Library/Homebrew/bintray.rb#L32
I'm not sure of its purpose, so I sadly cannot comment it. I'm happy deleting this line, and we can add it back if we find that we need it.

Reply:
If I were to guess, it was a hack for uploading and verifying (more likely the verifying step) macOS bottles running on a Linux CI runner.

Reply:
I've deleted this line. We can crib it from `bintray.rb` if needed.